assets/components/directDebit/directDebitActions.js

```diff
@@ -33,6 +38,55 @@ const updateAccountHolderName = (accountHolderName: string): Action =>
 const updateAccountHolderConfirmation = (accountHolderConfirmation: boolean): Action =>
   ({ type: 'DIRECT_DEBIT_UPDATE_ACCOUNT_HOLDER_CONFIRMATION', accountHolderConfirmation });
 
+const setDirectDebitFormError = (message: string): Action =>
+  ({ type: 'DIRECT_DEBIT_SET_FORM_ERROR', message });
+
+const resetDirectDebitFormError = (): Action =>
+  ({ type: 'DIRECT_DEBIT_RESET_FORM_ERROR' });
+
+
+function payDirectDebitClicked(callback: Function): Function {
+
+  return (dispatch: Function, getState: Function) => {
+
+    const {
+      bankSortCode,
+      bankAccountNumber,
+      accountHolderName,
+      accountHolderConfirmation,
+    } = getState().page.directDebit;
+
+    const isTestUser: boolean = getState().page.user.isTestUser || false;
+    const { csrf } = getState().page;
+
+    dispatch(resetDirectDebitFormError());
+
+    if (!accountHolderConfirmation) {
+      dispatch(setDirectDebitFormError('You need to confirm that you are the account holder.'));
+      return;
+    }
+
+    checkAccount(bankSortCode, bankAccountNumber, accountHolderName, isTestUser, csrf)
+      .then((response) => {
+        if (!response.accountValid) {
+          throw new Error('code1');
+        }
+      }).then(() => {
+        callback(undefined, bankAccountNumber, bankSortCode, accountHolderName);
+      }).catch((e) => {
+        let msg = '';
+        switch (e.message) {
+          case 'code1': msg = 'Your bank data is not ok, please check it and try again.';
```

Comment:
Are we going to get copy written for this in the final iteration?

Reply:
Yes, this is just me being "creative", this will not be exposed to the end user. It is behind a switch.